src/nectarchain/calibration/container/charge.py

```diff
@@ -124,140 +126,212 @@ def __init__(self,charge_hg,charge_lg,peak_hg,peak_lg,run_number,pixels_id,neven
+        self.trig_pattern_all = np.zeros(
+            (self.nevents, self.CAMERA.n_pixels, 4), dtype=bool
+        )
 
     @classmethod
-    def from_waveforms(cls,waveformContainer : WaveformsContainer,method : str = "FullWaveformSum",**kwargs) : 
-        """ create a new ChargeContainer from a WaveformsContainer
+    def from_waveforms(
+        cls,
+        waveformContainer: WaveformsContainer,
+        method: str = "FullWaveformSum",
+        **kwargs,
+    ):
+        """create a new ChargeContainer from a WaveformsContainer
         Args:
             waveformContainer (WaveformsContainer): the waveforms
-            method (str, optional): Ctapipe ImageExtractor method. Defaults to "FullWaveformSum".
+            method (str, optional): Ctapipe ImageExtractor method.
+            Defaults to "FullWaveformSum".
 
         Returns:
             chargeContainer : the ChargeContainer instance
         """
         log.info(f"computing hg charge with {method} method")
-        charge_hg,peak_hg = ChargeContainer.compute_charge(waveformContainer,constants.HIGH_GAIN,method,**kwargs)
-        charge_hg = np.array(charge_hg,dtype = np.uint16)
+        charge_hg, peak_hg = ChargeContainer.compute_charge(
+            waveformContainer, constants.HIGH_GAIN, method, **kwargs
+        )
+        charge_hg = np.array(charge_hg, dtype=np.uint16)
         log.info(f"computing lg charge with {method} method")
-        charge_lg,peak_lg = ChargeContainer.compute_charge(waveformContainer,constants.LOW_GAIN,method,**kwargs)
-        charge_lg = np.array(charge_lg,dtype = np.uint16)
+        charge_lg, peak_lg = ChargeContainer.compute_charge(
+            waveformContainer, constants.LOW_GAIN, method, **kwargs
+        )
+        charge_lg = np.array(charge_lg, dtype=np.uint16)
+
+        chargeContainer = cls(
+            charge_hg,
+            charge_lg,
+            peak_hg,
+            peak_lg,
+            waveformContainer.run_number,
+            waveformContainer.pixels_id,
+            waveformContainer.nevents,
+            waveformContainer.npixels,
+            method,
+        )
 
-        chargeContainer = cls(charge_hg,charge_lg,peak_hg,peak_lg,waveformContainer.run_number,waveformContainer.pixels_id,waveformContainer.nevents,waveformContainer.npixels ,method)
-        
         chargeContainer.ucts_timestamp = waveformContainer.ucts_timestamp
         chargeContainer.ucts_busy_counter = waveformContainer.ucts_busy_counter
         chargeContainer.ucts_event_counter = waveformContainer.ucts_event_counter
         chargeContainer.event_type = waveformContainer.event_type
         chargeContainer.event_id = waveformContainer.event_id
         chargeContainer.trig_pattern_all = waveformContainer.trig_pattern_all
 
-        return chargeContainer 
+        return chargeContainer
 
-
-    def write(self,path : Path,**kwargs) : 
+    def write(self, path: Path, bloc, start, end, **kwargs):
```

Comment:
I think `bloc` (`block` ?), `start` and `end` should have default values such that a whole run is processed and written if no value are provided for these arguments.